api/address.proto

```diff
@@ -13,16 +13,25 @@ message Address {
     // matching in FilterChainMatch. For clusters, an address may be either an
     // IP or hostname to be resolved via DNS.
     google.protobuf.StringValue address = 2;
-    oneof port {
+    oneof port_specifier {
```

Comment:
I get why we are doing this (need a single address that can be used to specify the outbound query), and then can also use it to describe the resolved address, but this feels strange to me. Could SRV be supported directly somehow in the resolver config vs. pushing it in here?